src/decimal.rs

```diff
@@ -3958,6 +3954,38 @@ impl ToPrimitive for Decimal {
     }
 }
 
+impl core::convert::TryFrom<f32> for Decimal {
+    type Error = crate::Error;
+
+    fn try_from(value: f32) -> Result<Self, Error> {
+        Self::from_f32(value).ok_or_else(|| Error::new("Failed to convert to Decimal"))
```

Comment:
I wasn't sure what message these errors should contain. Since From/ToPrimitive return Option, there's no information about what failed in particular.